src/thermal.cpp

```diff
@@ -151,25 +151,25 @@ ThermalScattering::calculate_xs(double E, double sqrtkT, int* i_temp,
 {
   // Determine temperature for S(a,b) table
   double kT = sqrtkT*sqrtkT;
-  int i;
-  if (settings::temperature_method == TemperatureMethod::NEAREST) {
-    // If using nearest temperature, do linear search on temperature
-    for (i = 0; i < kTs_.size(); ++i) {
-      if (std::abs(kTs_[i] - kT) < K_BOLTZMANN*settings::temperature_tolerance) {
-        break;
-      }
-    }
-  } else {
+  int i = 0;
+
+  if (kTs_.size() > 1) {
```

Comment:
Correct me if I'm wrong, but I think there is still a problem here for temperatures that are lower than the set of loaded cross section temperatures. For example, if there is a model temperature at 150 K and loaded cross sections at 200 K, 500 K, 800 K, and 900K then the entire loop below will run to the end of the loaded cross sections and we'll end up with and index indicating that our temperature is between 800 K and 900 K. The temperature _farthest_ from 150 K will be used based on the selection in the next section. From what I can tell this works just fine for temperatures higher than the highest loaded cross section however.

We'll want to add a check for temps below `kTs_[0]` and skip the loop.